### PR TITLE
[B] Allow text titles to be formatted

### LINKED ISSUE #1563: Markdown doesn't render in title fields
Markdown doesn't render as expected in the project title field nor in the the text title field:

project-title
<hr/>
text-title

It does render as expected in the resource title field, but having an asterisk as the first character seems to break the resource, serving [only a blank page](https://edge.manifoldapp.org/projects/this-is-a-title-within-a-title/resource/title-in-title).

The samples here come from a [dummy project on Edge](https://edge.manifoldapp.org/projects/this-is-a-title-within-a-title).

### PR DESCRIPTION
**Pull request by SMaxOwok**
Fixes #1563

### CODE REVIEW

File: api/app/models/text.rb [hunk 1]
```diff
@@ -199,6 +205,21 @@ def to_s
 
   private
 
+  def title_changed?
+    reload_title
```
Comment (SMaxOwok) [inline api/app/models/text.rb]: Are there any suggestions on how to do this that doesn't require reloading every time _or_ that can be a little smarter about when to reload?  The `title` method get the value from the main title association, which will have changed during `title=`.  

Right now, to compare the old value with the current I need to refresh that association.

**Comment by SMaxOwok** [comment]
@zdavis updated to delegate formatted values to the main text title association

**Comment by zdavis** [comment]
Snapshot failing here:
```
FAIL  src/containers/backend/Text/__tests__/Wrapper-test.js
  ● Backend Text Wrapper Container › renders correctly
    expect(value).toMatchSnapshot()
    
    Received value does not match stored snapshot 1.
    
    - Snapshot
    + Received
    
    @@ -14,11 +14,11 @@
              >
                <i
                  aria-hidden="true"
                  className="manicon manicon-arrow-left"
                />
    -           Rowan Test
    +           Back
              </a>
            </div>
          </nav>
          <div
            className="wrapper"
      42 |   });
      43 | 
    > 44 |   it("doesn't render to null" () => {
      45 |     let tree = component.toJSON();
      46 |     expect(tree).not.toBe(null);
      47 |   });
      
      at Object.<anonymous> (src/containers/backend/Text/__tests__/Wrapper-test.js:44:6)

```

**Comment by SMaxOwok** [comment]
@zdavis updated

**Pull request merged**